specifications/api/swagger.yaml

```diff
@@ -892,10 +892,35 @@ x-delegationTarget: &delegationTarget
 x-addressIndex: &addressIndex
   type: number
   minimum: 0
-  maximum: 4294967296 # 2 ^ 32, whole range (soft and hard) for random addresses.
+  # 2 ^ 32 - 1, whole range (soft and hard) for random addresses:
+  maximum: 4294967295
```

Comment:
the question for me is whether to differentiate soft and hardened index
we need to remember that hardened one can assume values 
```
<0x80000000, maxBound @Word32>
```
and soft index can be 
<0, 0x80000000 -1>

Reply:
> If we use the same index for both soft and hardened then the index can be between
> ```<0, 0x80000000 - 1> = <0,2147483648-1>```
> So the maximum is wrong! BUT, I remember about `WholeDomain` !!! and here we need this maximum in `x-addressIndex`. So maybe introduce `x-addressRelativeIndex` instead.

Using a relative index sounds like a nice solution here.

If we use the following `x-relativeAddressIndex` definition:

```yaml
x-relativeAddressIndex: &relativeAddressIndex 
  type: number
  minimum: 0  
  # 2 ^ 31 - 1
  maximum: 2147483647 
  description: |  
    A relative address derivation index.      
```

This means that our API type can also be very simple:
```hs
newtype ApiRelativeAddressIndex = ApiRelativeAddressIndex 
    { unApiRelativeAddressIndex :: Word31 
    } deriving (Bounded, Enum, Eq, Generic, Show)
```

Parsing an index value is now very simple, as we just need to check that it lies within the range `[0 ... 2^31 - 1]`.